examples/ui/size_constraints.rs

```diff
@@ -358,38 +350,39 @@ fn update_buttons(
 }
 
 fn update_radio_buttons_colors(
-    mut event_reader: EventReader<ButtonActivatedEvent>,
+    mut click_events: EventReader<Clicked>,
     button_query: Query<(Entity, &Constraint, &Interaction)>,
     mut color_query: Query<&mut BackgroundColor>,
     mut text_query: Query<&mut Text>,
     children_query: Query<&Children>,
 ) {
-    for &ButtonActivatedEvent(button_id) in event_reader.read() {
-        let target_constraint = button_query.get_component::<Constraint>(button_id).unwrap();
-        for (id, constraint, interaction) in button_query.iter() {
-            if target_constraint == constraint {
-                let (border_color, inner_color, text_color) = if id == button_id {
-                    (ACTIVE_BORDER_COLOR, ACTIVE_INNER_COLOR, ACTIVE_TEXT_COLOR)
-                } else {
-                    (
-                        INACTIVE_BORDER_COLOR,
-                        INACTIVE_INNER_COLOR,
-                        if matches!(interaction, Interaction::Hovered) {
-                            HOVERED_TEXT_COLOR
-                        } else {
-                            UNHOVERED_TEXT_COLOR
-                        },
-                    )
-                };
+    for &Clicked(button_id) in click_events.read() {
+        if let Ok(target_constraint) = button_query.get_component::<Constraint>(button_id) {
```

Comment:
I would advice to not change unrelated places to keep diffs nice.

Reply:
If you're referring to the `if let` I added, the code crashes if I don't have that, unclear as to why.

Reply:
Oh, of course, it makes sense because now we listen for all clicks. Sorry :)